client/karma.js

```diff
@@ -259,24 +259,26 @@ function Karma (socket, iframe, opener, navigator, location, document) {
   }
 
   socket.on('execute', function (cfg) {
-    // reset startEmitted and reload the iframe
-    startEmitted = false
-    self.config = cfg
-    // if not clearing context, reloadingContext always true to prevent beforeUnload error
-    reloadingContext = !self.config.clearContext
-    navigateContextTo(constant.CONTEXT_URL)
-
-    if (self.config.clientDisplayNone) {
-      [].forEach.call(document.querySelectorAll('#banner, #browsers'), function (el) {
-        el.style.display = 'none'
-      })
-    }
+    // Delay our navigation to the next event in case the clearContext has not completed.
+    setTimeout(function allowClearContextToComplete () {
+      // reset startEmitted and reload the iframe
+      startEmitted = false
+      self.config = cfg
+
+      navigateContextTo(constant.CONTEXT_URL)
+
+      if (self.config.clientDisplayNone) {
+        [].forEach.call(document.querySelectorAll('#banner, #browsers'), function (el) {
```

Comment:
I think Array.from(document.querySelectorAll()).forEach(...) is more idiomatic, but I understand that this section is just change in indentation so I'll let you decide if we want to change this line in this PR

Reply:
Or even `for (const el of document.querySelectorAll('#banner, #browsers')) { ... }`.

Reply:
While I agree with these readability comments, I am reluctant to apply the change as this would break us on IE11
https://caniuse.com/#search=array.from
https://caniuse.com/#search=for...of